x/precisebank/types/fractional_balance.go

```diff
@@ -7,9 +7,20 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// maxFractionalAmount is the largest valid value in a FractionalBalance amount.
-// This is for direct internal use so that there are no extra allocations.
-var maxFractionalAmount = sdkmath.NewInt(1_000_000_000_000).SubRaw(1)
+var (
+	// conversionFactor is used to convert the fractional balance to integer
+	// balances.
+	conversionFactor = sdkmath.NewInt(1_000_000_000_000)
+	// maxFractionalAmount is the largest valid value in a FractionalBalance amount.
+	// This is for direct internal use so that there are no extra allocations.
+	maxFractionalAmount = conversionFactor.SubRaw(1)
```

Comment:
I wonder if we need maxFractionalAmount at all?

It seems simpler and represents how we'd write it on paper if we validate `0 <= amount < C` rather than `0 <= amount <= C - 1`.